stdlib/domain.mli

```diff
@@ -67,7 +67,7 @@ val at_exit : (unit -> unit) -> unit
     the function most recently added with [at_exit] is called first. An example:
 
     {[
-let temp_file_key = Domain.DLS.new_key (fun _ ->
+let temp_file_key = Thread_local_storage.new_key (fun _ ->
   let tmp = snd (Filename.open_temp_file "" "") in
   Domain.at_exit (fun () -> close_out_noerr tmp);
   tmp)
```

Comment:
The example isn't broken (well it is, since there is a typo). A temporary file descriptor will now be created per _thread_, which will be closed when the domain that the thread is running on exits. It is not what you will want ideally, but the file does get closed.

Reply:
(I think the typo is that `new_key` should now be `make`.)

When I wrote the comment I thought that the domain would only close one file descriptor on exit (which is broken), but in fact the cumulative behavior of `Domain.at_exit` (adding a new callback instead of replacing the callback) means, as you point out, that all descriptors will be closed on domain termination.

I still consider that this example is not right, however. Consider in particular the case of a single domain that regularly spawns short-lived threads (to execute a blocking call, for example). What the user needs is to be able to keep a number of file descriptors alive that corresponds to the number of threads that are alive. Instead the number of file descriptors open will grow in an unbounded way until program termination. This is not just "no ideal", in practice it is not acceptable for this workload.

Remark: putting a finalizer on the value produced by the initialisation function would work better¹ than using `Domain.at_exit`. If we don't want to have `Thread.at_exit`, maybe we should rewrite the example to use a finalizer?

¹: I assume that thread termination clears the TLS array of the thread, but I am not sure that this is the case with the current PR.